### DISCUSSION TITLE
How to not-use Composer

### DISCUSSION THREAD

**Discussion by Solhdeck** [PHPOffice/PhpSpreadsheet#3747] [Q&A] [answered]
I'd like to use this tool, but I only can upload FTP files to my server, I don't have console access or anything like that.

How can I do for reading xlsx files?

I don't know if there is any option to use Composer from a php file or something, but I don't have access to the php.ini file or the console in the server.

About using Composer in local... I don't know exactly what PHP version is installed in the server (8 something) or... if there is anything else to have on count (like what modules are loaded).

There is really no other option to use this tool?
Does someone have the autoload.php for any 8.x version of php? or just the latest current version?

Thanks

**Comment by MarkBaker** [comment, marked as answer]
If you FTP files from your local development machine to your server, then you run composer on your local develop machine and it will retrieve all the files that you need to your local development machine. You can then FTP them to your server.

As a developer, you should always know what version of PHP you are developing for, and what version is running on your server. Use FTP to upload a file containing a `phpinfo()` command and then call that script (you can delete it again afterwards), which will tell you what version you have installed, and what modules are loaded. If you don't know, how do you know what features of PHP you can use and what you can't use?
You should always be using the same version of PHP with the same modules for development that are on your server. How else do you test your code?

[PhpSpreadsheet without Composer](https://github.com/PHPOffice/PhpSpreadsheet/discussions/1969)
